modules/apps/27-interchain-accounts/module.go

```diff
@@ -147,9 +159,24 @@ func (am AppModule) OnChanOpenInit(
 	counterparty channeltypes.Counterparty,
 	version string,
 ) error {
-	return am.keeper.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, version)
+	if err := am.keeper.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, version); err != nil {
+		return err
+	}
+
+	// issue capability for connected authentication module
+	appCap, err := am.scopedKeeper.NewCapability(ctx, types.AppCapabilityName(portID, channelID))
+	if err != nil {
+		return sdkerrors.Wrap(err, "could not create capability for underlying application")
+	}
+
+	// call underlying app's OnChanOpenInit callback with the appVersion
+	return am.app.OnChanOpenInit(ctx, order, connectionHops, portID, channelID,
```

Comment:
Question: If we go with the middleware approach, does it mean that every module that uses interchain accounts (helper/auth module) will have to be an IBC application (i.e. has the callbacks setup)? Currently, the helper modules aren't necessarily IBC applications.

Reply:
The helper modules will need to implement the IBC App module interface, but they can choose to do no custom logic for these functions

I would consider all helper modules IBC apps even if they aren't directly so, they are causing an IBC packet to be sent and performing logic upon its acknowledgement or timeout

I've been referring to helper modules as authentication modules. Do you have a preferred term, I find helper to be a little vague

Reply:
Yeah, I dislike the term `helper` module a lot. Authentication module sounds good for the time being :+1: 

Good point about doing something on ack/timeout for the packets (the hooks approach previously was a little convoluted). So basically when we go with this approach (ica middleware + auth module implementing IBC interface) each ica-auth module can implement custom logic on ack/timeout via the standard IBC App interface?

Reply:
> So basically when we go with this approach (ica middleware + auth module implementing IBC interface) each ica-auth module can implement custom logic on ack/timeout via the standard IBC App interface?

yup, custom logic can be implemented on:
- channel handshake init and ack
- send packet
- ack
- timeout

(controller side logic only)